### ISSUE TITLE
Do not fail on vulnerabilities that have no patch

### ISSUE THREAD

**Issue by JorritSalverda** [estafette/estafette-extension-npm-audit#10]
Currently the npm-audit extension fails on a vulnerability like the one below that doesn't have a patch available yet. Since this is not actionable it makes no sense to fail the audit:

```
=== npm audit security report ===                        
                                                                                
                                                                                
                                 Manual Review                                  
             Some vulnerabilities require your attention to resolve             
                                                                                
          Visit https://go.npm.me/audit-guide for additional guidance           
                                                                                
                                                                                
  Low             Denial of Service                                             
                                                                                
  Package         node-sass                                                     
                                                                                
  Patched in      No patch available                                            
                                                                                
  Dependency of   node-sass [dev]                                               
                                                                                
  Path            node-sass                                                     
                                                                                
  More info       https://npmjs.com/advisories/961                              
                                                                                
found 1 low severity vulnerability in 11159 scanned packages
  1 vulnerability requires manual review. See the full report for details.

Failed checking vulnerabilities
Failed with exit code: 1
```

**Comment by JorritSalverda** [comment]
Running `npm audit --json` outputs the following:

```json
{
  "actions": [
    {
      "action": "review",
      "module": "node-sass",
      "resolves": [
        {
          "id": 961,
          "path": "node-sass",
          "dev": true,
          "bundled": false,
          "optional": false
        }
      ]
    }
  ],
  "advisories": {
    "961": {
      "findings": [
        {
          "version": "4.13.0",
          "paths": [
            "node-sass"
          ]
        }
      ],
      "id": 961,
      "created": "2019-06-12T13:44:17.616Z",
      "updated": "2020-01-13T20:16:49.738Z",
      "deleted": null,
      "title": "Denial of Service",
      "found_by": {
        "link": "",
        "name": "Alexander Jordan",
        "email": ""
      },
      "reported_by": {
        "link": "",
        "name": "Alexander Jordan",
        "email": ""
      },
      "module_name": "node-sass",
      "cves": [],
      "vulnerable_versions": ">=0.0.0",
      "patched_versions": "<0.0.0",
      "overview": "All versions of `node-sass` are vulnerable to Denial of Service (DoS). Crafted objects passed to the `renderSync` function may trigger C++ assertions in `CustomImporterBridge::get_importer_entry` and `CustomImporterBridge::post_process_return_value` that crash the Node process. This may allow attackers to crash the system's running Node process and lead to Denial of Service.",
      "recommendation": "No fix is currently available. Consider using an alternative package until a fix is made available.",
      "references": "",
      "access": "public",
      "severity": "low",
      "cwe": "CWE-400",
      "metadata": {
        "module_type": "",
        "exploitability": 2,
        "affected_components": ""
      },
      "url": "https://npmjs.com/advisories/961"
    }
  },
  "muted": [],
  "metadata": {
    "vulnerabilities": {
      "info": 0,
      "low": 1,
      "moderate": 0,
      "high": 0,
      "critical": 0
    },
    "dependencies": 63,
    "devDependencies": 11096,
    "optionalDependencies": 194,
    "totalDependencies": 11159
  },
  "runId": "2804a822-6626-4817-8e6f-44a8a73aaab3"
}
```

**Comment by JorritSalverda** [comment]
I'd still like to see a json structure in case there is a patch to see if a patch property is in the findings. Otherwise we need to look at the `patched_versions` property at the advisory level and ensure it's not `"<0.0.0"` or something like that.

**Comment by vitorarins** [comment]
@JorritSalverda does this PR #9 fixes this problem?
On the PR I change the behavior to look for actions instead of findings, so I think this is what we want.

**Comment by JorritSalverda** [comment]
A day later with a patch being available the json looks like

```json
{
  "actions": [
    {
      "action": "update",
      "resolves": [
        {
          "id": 961,
          "path": "node-sass",
          "dev": true,
          "optional": false,
          "bundled": false
        }
      ],
      "module": "node-sass",
      "target": "4.13.1",
      "depth": 1
    }
  ],
  "advisories": {
    "961": {
      "findings": [
        {
          "version": "4.13.0",
          "paths": [
            "node-sass"
          ]
        }
      ],
      "id": 961,
      "created": "2019-06-12T13:44:17.616Z",
      "updated": "2020-01-16T13:57:56.161Z",
      "deleted": null,
      "title": "Denial of Service",
      "found_by": {
        "link": "",
        "name": "Alexander Jordan",
        "email": ""
      },
      "reported_by": {
        "link": "",
        "name": "Alexander Jordan",
        "email": ""
      },
      "module_name": "node-sass",
      "cves": [],
      "vulnerable_versions": ">=3.3.0 <4.13.1",
      "patched_versions": ">=4.13.1",
      "overview": "Affected versions of `node-sass` are vulnerable to Denial of Service (DoS). Crafted objects passed to the `renderSync` function may trigger C++ assertions in `CustomImporterBridge::get_importer_entry` and `CustomImporterBridge::post_process_return_value` that crash the Node process. This may allow attackers to crash the system's running Node process and lead to Denial of Service.",
      "recommendation": "Upgrade to version 4.13.1 or later",
      "references": "",
      "access": "public",
      "severity": "low",
      "cwe": "CWE-400",
      "metadata": {
        "module_type": "",
        "exploitability": 2,
        "affected_components": ""
      },
      "url": "https://npmjs.com/advisories/961"
    }
  },
  "muted": [],
  "metadata": {
    "vulnerabilities": {
      "info": 0,
      "low": 1,
      "moderate": 0,
      "high": 0,
      "critical": 0
    },
    "dependencies": 63,
    "devDependencies": 11096,
    "optionalDependencies": 194,
    "totalDependencies": 11159
  },
  "runId": "076c2fa3-8766-4104-9709-a8c03350029e"
}
```

**Comment by JorritSalverda** [comment]
> @JorritSalverda does this PR #9 fixes this problem?
> On the PR I change the behavior to look for actions instead of findings, so I think this is what we want.

Only for `"action": "update"`, we should ignore it if we see `"action": "review"`.

**Comment by JorritSalverda** [comment]
@vitorarins I've added some changes to your PR to check the action; if it's `review` it doesn't fail. Not entirely sure that review isn't used in case of patchable vulnerabilities though.

**Comment by JorritSalverda** [comment]
Fixed by #9
